stackhpc_monasca_agent_plugins/checks/slurm.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2018 StackHPC Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import copy
+import logging
+import re
+
+import monasca_agent.collector.checks as checks
+from monasca_agent.common.util import timeout_command
+
+
+log = logging.getLogger(__name__)
+
+_METRIC_NAME_PREFIX = "slurm"
+_METRIC_NAME = "job_status"
+
+_SLURM_LIST_JOBS_CMD = ['/usr/bin/scontrol', '-o', 'show', 'job']
+_SLURM_LIST_NODES_CMD = ['/usr/bin/scontrol', '-o', 'show', 'node']
+
+_SLURM_JOB_FIELD_REGEX = ('^JobId=([\d]+)\sJobName=((?i)[\w-]+\.[\w-]+)\sUserI'
+                          'd=([\w-]+\([\w-]+\)) GroupId=([\w-]+\([\w-]+\))\s.*'
+                          'JobState=([\w]+)\s.*\sNodeList=(.*?)\s.*$')
+_SLURM_NODE_FIELD_REGEX = '^NodeName=(.*?)\s.*State=(.*?)\s.*$'
+_SLURM_NODE_SEQUENCE_REGEX = '^(.*)-\[(\d+-\d+)\]$'
```

Comment:
Slurm really needs an API.

Reply:
I suppose I could call the C API via ctypes or SWIG. I'm not sure how much time we wanted to spend on this. @oneswig ?

Reply:
I looked again at PySlurm, and Cython, perhaps the use of Cython would not be so bad afterall - I think it's mostly build time changes. It might be interesting to investigate at some point. We're probably best sticking with the command output parsing for now though.